fenicsprecice/adapter_core.py

```diff
@@ -95,7 +96,7 @@ def determine_function_type(input_obj):
     if isinstance(input_obj, FunctionSpace):  # scalar-valued functions have rank 0 is FEniCS
         if input_obj.num_sub_spaces() == 0:
             return FunctionType.SCALAR
-        elif input_obj.num_sub_spaces() == 2:
+        elif input_obj.num_sub_spaces() == 2 or input_obj.num_sub_spaces() == 3:
             return FunctionType.VECTOR
```

Comment:
Is it possible to add a consistency check with the dimensionality here? Or does this happen somewhere else? preCICE only allows 2d vectors for 2d simulations and 3d vectors for 3d simulations. To increase security and readability, I would propose something like (be careful: pseudocode!):

```
elif input_obj.num_sub_spaces() == 2:
    assert(precice.get_dimensions() == 2)
    return FunctionType.VECTOR
elif input_obj.num_sub_spaces() == 3:
    assert(precice.get_dimensions() == 3)
    return FunctionType.VECTOR

Reply:
The dimensionality check happens later in initialize:
https://github.com/precice/fenics-adapter/blob/80e541422955b948c3ef0626c4cbf633291e10ce/fenicsprecice/fenicsprecice.py#L365-L366

so the user will not go beyond this point with a dimension mismatch. Is this not enough? In the function `determine_function_type()` we only want to find out if we are dealing with a scalar or vector function.